### PR TITLE
More opts for http request

### PR DESCRIPTION
**Pull request by HJianBo**

### CODE REVIEW

File: etc/emqx_auth_http.conf [hunk 1]
```diff
@@ -2,13 +2,47 @@
...
+## Connection time-out time, used during the initial request
+## when the client is connecting to the server
+##
+## Value: Duration
+##
+## Default is the value of option timeout
+## auth.http.request.connect_timout = 5s
+
+
+##
```
Comment (Gilbert-Wong) [inline etc/emqx_auth_http.conf]: More description for this config entry.

File: src/emqx_auth_http_cli.erl [hunk 2]
```diff
@@ -25,20 +23,21 @@
...
 
-request_(Method, Req, HTTPOpts, Opts, Times) ->
+request_(Method, Req, HTTPOpts, Opts, RetryOpts = #{times := Times,
+                                                    interval := Interval}) ->
     %% Resend request, when TCP closed by remotely
     case httpc:request(Method, Req, HTTPOpts, Opts) of
-        {error, socket_closed_remotely} when Times < 3 ->
-            timer:sleep(trunc(math:pow(10, Times))),
-            request_(Method, Req, HTTPOpts, Opts, Times+1);
+        {error, socket_closed_remotely} when Times > 0 ->
```
Comment (HJianBo) [inline src/emqx_auth_http_cli.erl]: {error,_} ->
    ....

File: etc/emqx_auth_http.conf [hunk 3]
```diff
@@ -2,13 +2,55 @@
...
+## Default: 0.1s
+auth.http.request.retry_interval = 0.1s
+
+## The 'Exponential Backoff' mechanism for re-sending request. The actually
+## re-send time interval is `interval * backoff ^ times`
+##
+## Value: float
+##
+## Default: 1.5
+auth.http.request.retry_backoff = 1.5
```
Comment (HJianBo) [inline etc/emqx_auth_http.conf]: 2 is better

File: etc/emqx_auth_http.conf [hunk 4]
```diff
@@ -2,13 +2,55 @@
...
+##
+## Default: 3
+auth.http.request.retry_times = 3
+
+## The interval for re-sending the http request
+##
+## Value: Duration
+##
+## Default: 0.1s
+auth.http.request.retry_interval = 0.1s
```
Comment (HJianBo) [inline etc/emqx_auth_http.conf]: 1s

File: test/emqx_auth_http_SUITE.erl [hunk 5]
```diff
@@ -67,31 +72,30 @@ set_special_configs(_App) ->
...
     deny = emqx_access_control:check_acl(SuperUser, publish, <<"anytopic">>),
 
-    User1 = #{client_id => <<"client1">>, username => <<"testuser">>, peername => {{127,0,0,1}, 2981}, zone => external},
-    UnIpUser1 = #{client_id => <<"client1">>, username => <<"testuser">>, peername => {{192,168,0,4}, 2981}, zone => external},
-    UnClientIdUser1 = #{client_id => <<"unkonwc">>, username => <<"testuser">>, peername => {{127,0,0,1}, 2981}, zone => external},
-    UnnameUser1= #{client_id => <<"client1">>, username => <<"unuser">>, peername => {{127,0,0,1}, 2981}, zone => external},
+    User1 = ?USER(<<"client1">>, <<"testuser">>, {{127,0,0,1}, 1883}, {{127,0,0,1}, 2981}, external),
+    UnIpUser1 = ?USER(<<"client1">>, <<"testuser">>, {{127,0,0,1}, 1883}, {{192,168,0,4}, 2981}, external),
+    UnClientIdUser1 = ?USER(<<"unkonwc">>, <<"testuser">>, {{127,0,0,1}, 1883}, {{127,0,0,1}, 2981}, external),
+    UnnameUser1= ?USER(<<"client1">>, <<"unuser">>, {{127,0,0,1}, 1883}, {{127,0,0,1}, 2981}, external),
```
Comment (Gilbert-Wong) [inline test/emqx_auth_http_SUITE.erl]: I suppose it is more clear if you do not use `macro` like `?USER` and directly write map structure here. Anyway, it's a matter of preference. Don't care about it.

**Pull request merged**